examples/generate_type_stubs.py

```diff
@@ -0,0 +1,469 @@
+Run this to regenerate type stubs after adding new operations.
+
+Usage:
+    python examples/generate_type_stubs.py
+
+RATIONALE:
+---------
+Type stubs are generated rather than handwritten because:
+1. pyvips is a binding around libvips C library, which evolves independently
+2. libvips has 300+ operations with complex signatures
+3. New operations are added to libvips frequently
+4. Generation uses same introspection mechanism as doc generation
+5. Ensures stubs stay synchronized with available operations
+
+This approach follows the existing pattern used for:
+- Enum generation (examples/gen-enums.py)
+- Documentation generation (pyvips.Operation.generate_sphinx_all())
+"""
+
+import typing
+from typing import Any, Optional, Union
+
+import pyvips
+from pyvips import (
+    GValue,
+    Introspect,
+    type_map,
+    type_from_name,
+    nickname_find,
+    at_least_libvips,
+)
+from pyvips import ffi, Error, _to_bytes, _to_string
+
+
+def gtype_to_python_type(gtype: int) -> str:
+    """Map a gtype to Python type annotation string."""
+    fundamental = pyvips.gobject_lib.g_type_fundamental(gtype)
+
+    if fundamental == GValue.genum_type:
+        name = pyvips.type_name(gtype)
+        if name.startswith("Vips"):
+            name = name[4:]
+        return f"Union[str, {name}]"
+
+    type_mapping = {
+        GValue.gbool_type: "bool",
+        GValue.gint_type: "int",
+        GValue.guint64_type: "int",
+        GValue.gdouble_type: "float",
+        GValue.gstr_type: "str",
+        GValue.refstr_type: "str",
+        GValue.gflags_type: "int",
+        GValue.gobject_type: "GObject",
+        GValue.image_type: "Image",
+        GValue.array_int_type: "list[int]",
+        GValue.array_double_type: "list[float]",
+        GValue.array_image_type: "list[Image]",
+        GValue.blob_type: "str",
+        GValue.source_type: "Source",
+        GValue.target_type: "Target",
+    }
+
+    if gtype in type_mapping:
+        return type_mapping[gtype]
+    if fundamental in type_mapping:
+        return type_mapping[fundamental]
+    return "Any"
+
+
+PYTHON_KEYWORDS = {
+    "in",
+    "min",
+    "max",
+    "type",
+    "class",
+    "def",
+    "return",
+    "import",
+    "from",
+    "as",
+    "if",
+    "else",
+    "elif",
+    "for",
+    "while",
+    "break",
+    "continue",
+    "pass",
+    "raise",
+    "try",
+    "except",
+    "finally",
+    "with",
+    "lambda",
+    "and",
+    "or",
+    "not",
+    "is",
+    "None",
+    "True",
+    "False",
+}
+
+
+def escape_parameter_name(name: str) -> str:
+    """Escape Python keywords in parameter names."""
+    if name in PYTHON_KEYWORDS:
+        return f"{name}_"
+    return name
+
+
+def generate_method_signature(operation_name: str) -> str:
+    """Generate type signature for an operation method."""
+    intro = Introspect.get(operation_name)
+
+    if (intro.flags & 4) != 0:  # _OPERATION_DEPRECATED
```

Comment:
Is this correct? `VipsOperationFlags` uses 8 for `VipsOperationFlags`. I think you're removed all uncacheable operations instead.